### ISSUE TITLE
Input area loses focus after command execution, seemingly at random

### ISSUE THREAD

**Issue by uxinnuendo** [warpdotdev/Warp#3729]
### Describe the bug

When entering any command (usually I'm just running git commands), often the input area loses focus after the command execution completes, and the input area must be manually reselected to continue executing additional commands.

The Warp window is still in focus (as Mac toolbar), but there is no active "section" and all subsequent key presses have no effect until one of the window sections is manually selected again.

Currently I only have one customisation: 
- Theme installed "Dracula"

There is no discernable pattern I have noticed:
- occurs both when an error code is returned, or command is successful (exit code 0)
- can occur on any type of git command

### To reproduce

1. $ git log
2. select and copy SHA from log view
3. $ git diff {file}
4. command completes with git diff shown, and the input area has lost focus
5. manually reselect 

Again, this is not a reliable reproduction test.

### Expected behavior

Maintain focus on Warp input area, unless focus was changed by the user during command execution.

### Operating system

MacOS

### Operating system and version

Ventura 13.4

### Shell Version

zsh 5.9 (x86_64-apple-darwin22.0)

### Current Warp version

v0.2023.09.26.08.09.stable_00

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Additional context

If no one else is experiencing this issue then I'd be happy to close this and try some other things (upgrade OS etc)

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by dannyneira** [comment]
Hi @uxinnuendo Sorry to hear about this issue. It's been difficult to reproduce on my end, do you have the blinking cursor disabled by any chance? Also, as a workaround, you can press `CMD-L` to quickly focus on the input w/o clicking.

To anyone else facing this issue, please add a :+1: to the original post at the top or comment with your details, and subscribe if you'd like to be notified.

**Comment by cforcey** [comment]
Yeah, I seem to be losing the focus on the command line app I am working in pretty often. I also find the re-focus behavior when I click back on warp to be confusing.  I usually check something about a command I am writing in a browser, return to warp, and hit up arrow expecting to get the history of the command line where I was typing scrolling.  Instead, I move the blue focus bar up and down the blocks which is rarely what I want.

**Comment by simonc** [comment]
Hi there 🙂 

Same here, the loss of focus when swapping apps can be very frustrating as you sometimes need to be precise in where you click to be able to type something again.

This is especially true when inside an interactive session like `irb` or `psql`.  Sadly CMD-L does do the trick here as the menu entry is greyed out. The behavior that would feel "natural" would be to focus the input when clicking anywhere in the area below the input or simply when starting to type.

I understand it might seem like a simple behavior but given all the rest is probably not 😅 

Thanks <3

**Comment by arusa** [comment]
When I open a tab and enter for example `ssh myserver` and then I immediately switch to another tab and try to enter for example `ssh mysecondserver` I lose focus and can't finish typing as soon as the first tab finished loading the bash prompt. The first tab is stealing the focus from my second tab although I can't see it. The second tab is still in focus, but the cursor is gone and I can't type anymore until I click the input area.

**Comment by lumberman** [comment]
same problem. very confusing. No matter what I do, I can't get back into the process.
